3-canali-standard-c/test.c

```diff
@@ -1,10 +1,15 @@
 #include <stdio.h>
+#include <ctype.h>
 
-int main() {
-    fprintf(stdout, "Test output\n");
-    fprintf(stderr, "Test errore\n");
-
-    printf("Leggo il primo carattere di input...\n"); // equivalente a fprintf(stdin...)
-    char c = fgetc(stdin);
-    printf("Hai scritto %c\n", c);
+int main () {
+   int i = 0;
+   char str[] = "ciao mondo";
+   
+   while(str[i]) {
+      putchar (toupper(str[i]));
+      i++;
+   }
+   
+   return(0);
 }
+
```

Comment:
Il programma stampa `CIAO MONDO` in output. Tuttavia, il programma dovrebbe stampare il testo che ottiene in input, trasformandolo in maiuscolo. Per leggere i singoli caratteri dal canale di input si può sfruttare la funzione `char c = getc(stdin)` (ad esempio), che legge un singolo carattere) e poi stampare quel carattere utilizzando `putchar(toupper(c))` (come è stato fatto).